### DISCUSSION TITLE
How to start the dbeaver with params

### DISCUSSION THREAD

**Discussion by janiu-001** [dbeaver/dbeaver#14454] [Q&A] [answered]
Hi expert,
1. Command line parameters might be passed directly to dbeaver[.exe] executable

How can i pass the params, can you give an example?

2. In Windows you can also use dbeaver-cli.exe executable (it does not spawn a new window so you can see the output messages).
Please also share where i can find the dbeaver-cli.exe , it is better to give a example also

3. Also, you can add parameters in the dbeaver.ini configuration file - in the beginning of the file and each parameter on its own line.
It is very appreciated to give an example also

Can you also share where the code to load the params for the first and the third way?

**Comment by nilawafer** [comment]
`dbeaver -con "name=$NAME|driver=PostgreSQL|host=$HOST|port=$PORT|database=$DATABASE|user=$USER|password=$PASSWORD"`
This works for me in the command line on mac, with $NAME being the name you want your connection to have (anything), $HOST is your host, you can change the driver to be whatever you need, the $PORT is like 5439 [for Redshift] or whatever standard port for your DMS, $USER is your username, and $PASSWORD is your password. Just make sure this is in your path: `/DBeaver.app/Contents/MacOS`. I have this saved as a shell function
This feature is broken in version 21.2.3, so I recommend downloading version 21.2.2

**Comment by janiu-001** [comment]
thanks very much for you response, and how i add the params with windows system, and can you provide the code how to reload the params

**Comment by ShadelessFox** [comment]
Hello @nilawafer,

This feature will be fixed within #14284 in DBeaver 21.2.4.

**Comment by ShadelessFox** [comment, marked as answer]
Hello @janiu-001,

All `dbeaver.exe`, `dbeaver-cli.exe`, and `dbeaver.ini` are located in DBeaver's installation folder.

You can locate it via its shortcut. Also you can specify these commands in the shortcut itself.
